components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentsTest.kt

```diff
@@ -486,75 +414,76 @@ class ExperimentsTest {
                     Experiment.Branch("branch3", 1)
                 )
             )
-        ))
-
+        )
         // This should force choosing branch 2 of the experiment on update.
         branchDiceRoll = 1
-        experimentsUpdater.updateExperiments()
+        experiments.onExperimentsUpdated(ExperimentsSnapshot(experimentsList, null))
```

Comment:
If we're removing the server mocking here, i assume we're testing that integration path (updater -> experiments -> correct result) in another unit test?

Reply:
I think that the `second update of experiments` test covers this integration path correctly.  I ended up removing some of the mocking here and calling the `onExperimentsUpdated()` directly to try and simplify an already complex test, and because trying to rely on the update while using the `mockContext` fails because the WorkManager requires an actual `context` to run and it doesn't work out well to mix an match mock and real contexts.

Reply:
Fair enough on the simplifying overly complex tests. As long as we cover the path elsewhere i'm fine with it.